controller/scrape.go

```diff
@@ -40,66 +40,95 @@ var numericStatus = map[string]float64{
+	ClusterFailedExperiments.WithLabelValues().Set(fail)
+	ClusterTotalExperiments.WithLabelValues().Set(total)
+}
+func setEngineChaosMetrics(engineDetails ChaosEngineDetail) {
+	EngineTotalExperiments.WithLabelValues(engineDetails.Namespace, engineDetails.Name).Set(engineDetails.TotalExp)
+	EnginePassedExperiments.WithLabelValues(engineDetails.Namespace, engineDetails.Name).Set(engineDetails.PassedExp)
+	EngineFailedExperiments.WithLabelValues(engineDetails.Namespace, engineDetails.Name).Set(engineDetails.FailedExp)
+}
 
-	// Set default values on the chaosResult map before populating w/ actual values
-	spec := ChaosResultSpec{ExporterSpec: exporterSpec, ChaosExperimentList: chaosExperimentList}
-	chaosResultMap := setChaosResultValue(clientSet, spec)
+func getExperimentMetricsFromEngine(chaosEngine *litmuschaosv1alpha1.ChaosEngine) (float64, float64, float64) {
+	var total, passed, failed float64
+	passed = 0
+	failed = 0
+	expStatusList := chaosEngine.Status.Experiments
+	total = float64(len(expStatusList))
+	for i, v := range expStatusList {
+		verdictFloat := getValueFromVerdict(v.Verdict)
+		if verdictFloat == 3 {
+			passed++
+		} else if verdictFloat == 2 {
+			failed++
+		} else if verdictFloat == 1 {
+			defineRunningExperimentMetric(chaosEngine.Name, chaosEngine.Namespace, chaosEngine.Spec.Experiments[i].Name)
+		}
+	}
+	return total, passed, failed
 
-	chaosResult, StatusMap := calculateChaosResult(chaosResultMap)
-	fmt.Printf("%+v\n", StatusMap)
-	totalExpCount := float64(len(engine.Spec.Experiments))
+}
+func defineRunningExperimentMetric(engineName string, engineNamespace string, experimentName string) {
+	klog.V(2).Infof("Running Experiment Metrics: EnginaName: %v, EngineNamespace: %v, ExperimentName: %v", engineName, engineNamespace, experimentName)
+	RunningExperiment.WithLabelValues(engineNamespace, engineName, experimentName, fmt.Sprintf("%s-%s", engineName, experimentName)).Set(float64(1))
 
-	return totalExpCount, chaosResult.TotalPassedExp, chaosResult.TotalFailedExp, StatusMap, nil
 }
 
-//setChaosResultValue will populate the default value of chaos result
-func setChaosResultValue(clientSet *clientV1alpha1.Clientset, chaosResultSpec ChaosResultSpec) map[string]string {
-	chaosResultMap := make(map[string]string)
-	for _, test := range chaosResultSpec.ChaosExperimentList {
-		chaosResultName := fmt.Sprintf("%s-%s", chaosResultSpec.ExporterSpec.ChaosEngine, test)
-		testResultDump, err := clientSet.LitmuschaosV1alpha1().ChaosResults(chaosResultSpec.ExporterSpec.AppNS).Get(chaosResultName, metav1.GetOptions{})
-		if err != nil {
-			if strings.Contains(err.Error(), "not found") {
-				// lack of result cr indicates experiment not executed
-				chaosResultMap[test] = "not-executed"
-			}
-			continue
-		}
-		chaosResultMap[test] = testResultDump.Status.ExperimentStatus.Verdict
+func getValueFromVerdict(verdict string) float64 {
+	if verdict == "Pass" || verdict == "pass" {
+		return 3
```

Comment:
Let us factor in the non-executed experiments by associating it with a metric value?